Client/Frontend/Browser/TabManager.swift

```diff
@@ -412,7 +412,14 @@ extension TabManager : WKNavigationDelegate {
 
     func webView(webView: WKWebView, didFinishNavigation navigation: WKNavigation!) {
         hideNetworkActivitySpinner()
-        storeChanges()
+        // only store changes if this is not an error page
+        // as we current handle tab restore as error page redirects then this ensures that we don't
+        // call storeChanges unnecessarily on startup
+        if let url = webView.URL {
```

Comment:
I wonder if this also fixes the issues users are seeing where on app restore they are seeing localhost domains.